src/components/Graph.vue

```diff
@@ -1,14 +1,141 @@
 ﻿<template>
   <div id='graph'>
+    <Searchbar id="search" v-on:input="searchFilter"></Searchbar>
     <div id='visualizer'></div>
   </div>
 </template>
 
 <script lang='ts'>
 import * as d3 from 'd3'
 import axios from 'axios'
+import Searchbar from '@/components/Searchbar.vue'
 
-function calulateViewBoxValue (width: number, height: number, viewBoxScalar: number) {
+export default {
+  name: 'Graph',
+  components: { Searchbar },
+  data () {
+    return {
+      searchText: '',
+      servers: null,
+      routes: null,
+      svg: null
+    }
+  },
+  async mounted () {
+    await this.getData()
+
+    const width = window.innerWidth
+    const height = window.innerHeight
+    const viewBoxScalar = 0.5
+
+    // SVG canvas
+    this.svg = d3.select('#visualizer')
+      .append('div')
+      .append('svg')
+      // Responsive SVG needs these 2 attributes
+      .attr('preserveAspectRatio', 'xMinYMin meet')
+      .attr('viewBox', calculateViewBoxValue(width, height, viewBoxScalar))
+
+    this.drawGraph(this.servers, this.routes, false)
+  },
+  methods: {
+    // Runs every time an input is given to the search bar - searchText is the input
+    searchFilter (searchText) {
+      this.searchText = searchText
+
+      if (this.searchText === '') {
+        this.drawGraph(this.servers, this.routes, false)
+      } else {
+        this.drawGraph(this.servers, this.routes, true)
+      }
+    },
+    // Checks whether the current server name contains the given search text/input
+    isSearchMatch (serverName) {
+      if (serverName === null) {
+        return false
+      }
+      return serverName.toLowerCase().includes(this.searchText.toLowerCase()) // Boolean statement
+    },
+    // Fetches data from backend and updates data variables
+    async getData () {
+      const varzResponse = await axios.get('https://localhost:5001/nodes')
+      this.servers = varzResponse.data
+
+      const routezResponse = await axios.get('https://localhost:5001/links')
+      this.routes = routezResponse.data
```

Comment:
```suggestion
    async getData () {
      const varzResponse = await axios.get('https://localhost:5001/nodes')
      this.servers = varzResponse.data

      const routezResponse = await axios.get('https://localhost:5001/links')
      this.routes = routezResponse.data
```

We shouldn't put localhost in there directly.  I'd suggest making environment variables called something like `APP_URL` and `APP_PORT`, and then fetching from there